hack/test-localhub.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+#
+# This script provisions testing environment for local Hub using docker-compose
+# and execute tests.
+#
+# It requires Docker to be installed.
+
+# standard bash error handling
+set -o nounset # treat unset variables as an error and exit immediately.
+set -o errexit # exit immediately when a command fails.
+
+CURRENT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+REPO_ROOT_DIR=$(cd "${CURRENT_DIR}/.." && pwd)
+
+# shellcheck source=./hack/lib/utilities.sh
+source "${CURRENT_DIR}/lib/utilities.sh" || { echo 'Cannot load CI utilities.'; exit 1; }
+# shellcheck source=./hack/lib/const.sh
+source "${CURRENT_DIR}/lib/const.sh" || { echo 'Cannot load constant values.'; exit 1; }
+
+build_images() {
+  export DOCKER_TAG=$RANDOM
+  export DOCKER_REPOSITORY="local"
+
+  local make_targets=("build-test-image-localhub" "build-app-image-hub-js" "build-app-image-secret-storage-backend")
+  for make_target in "${make_targets[@]}"
+  do
+    cd "${REPO_ROOT_DIR}" && make "${make_target}"
+  done
+}
+
+main() {
+  if [[ "${BUILD_IMAGES:-"true"}" == "true" ]]; then
+    build_images
+  fi
+    
+  docker-compose -f "${REPO_ROOT_DIR}/test/localhub/docker-compose.yml" up --exit-code-from tests --force-recreate
```

Comment:
BTW please also make sure all components exit and clean up properly. I'm not sure if that was related to the tests, but after a day or so I executed `docker ps` and saw `hub-js` still running in background. Unfortunately, Please check that during your tests after doing the proposed changes 🙂

Reply:
I added `docker-compose down` in order to delete the images after executing the tests. Although, the above situation still should not happen in "normal" behavior and after tests, the container should stop.